networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -418,3 +419,49 @@ def descendants_at_distance(G, source, distance):
         current_distance += 1
 
     return current_layer
+
+
+@not_implemented_for("directed")
+def _bfs_with_marks(G, start_node, check_set):
+    """Breadth-first-search with markings.
+
+    Performs BFS starting from ``start_node`` and whenever a node
+    inside ``check_set`` is met, it is "marked". Once a node is marked,
+    BFS does not continue along that path. The resulting marked nodes
+    are returned.
+
+    Parameters
+    ----------
+    G : nx.Graph
+        An undirected graph.
+    start_node : node
+        The start of the BFS.
+    check_set : set
+        The set of nodes to check against.
+
+    Returns
+    -------
+    marked : set
+        A set of nodes that were marked.
+    """
+    visited = dict()
+    marked = set()
+    queue = []
+
+    visited[start_node] = None
+    queue.append(start_node)
+    while queue:
+        m = queue.pop(0)
+
+        for neighbr in G.neighbors(m):
```

Comment:
This is a very minor nit....
Our standard abbreviated variable name for a neighbor is `nbr`.  Not a big deal -- but what do you think?

Reply:
Agreed! Standardization removes cognitive complexity. These tips help for me towards becoming hopefully a highly efficient and productive contributor to networkx :)